### ISSUE TITLE
Combobox traps VO cursor within the popover

### ISSUE THREAD

**Issue by Anon-136** [ariakit/ariakit#4036]
### Current behavior

I am playing with the combobox component with MacOS VoiceOver.
I found a problem when VO cursor navigates from the listbox back to the input
it is moved back to focus a listitem in the listbox.

https://github.com/user-attachments/assets/39b65c38-2426-4667-a25d-40f01a558ff4

https://ariakit.org/components/combobox 
version 0.4.8

### Steps to reproduce the bug

Steps to reproduce are the same as video above
1. Open https://ariakit.org/previews/combobox
2. Open VO on MacOS
3. Expand the combobox with VO + Space
4. Navigate to the listbox with VO + right
5. Navigate inside the listbox with VO + Shift + down
6. Navigate out of the listbox with VO + Shift + up
7. Navigate back to combobox input with VO + left

### Expected behavior

It should navigate back to the input in order to edit the text.

For example:
https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-autocomplete-list/

https://github.com/user-attachments/assets/1bca7b7c-08f4-40b3-b008-1581e51b4c93

### Workaround

-

### Possible solutions

I noticed that the a property `data-focus-visible` is set to true in the listitem.
It might be the problem here.

![CleanShot 2567-08-15 at 14 59 26@2x](https://github.com/user-attachments/assets/a72eb9b4-6bed-4fd0-8a01-22dd43f18daa)

**Comment by diegohaz** [comment]
Thank you for the report.

> ### Steps to reproduce the bug
> Steps to reproduce are the same as video above
> 
> 1. Open https://ariakit.org/previews/combobox
> 2. Open VO on MacOS
> 3. Expand the combobox with VO + Space
> 4. Navigate to the listbox with VO + right
> 5. Navigate inside the listbox with VO + Shift + down
> 6. Navigate out of the listbox with VO + Shift + up
> 7. Navigate back to combobox input with VO + left

I followed these instructions on macOS Sonoma 14.2 with Chrome 127 and couldn't reproduce the issue. The combobox input is correctly focused when using VO keys. Since Arc also uses Chromium, I'd expect similar behavior. Could you provide more details so I can try and reproduce it?

> ### Expected behavior
> It should navigate back to the input in order to edit the text.

Nonetheless, in your video, VO announces the combobox input when you navigate back to it, even though the virtual cursor is visually on the option. You should be able to edit the input text.

**Comment by Anon-136** [comment]
> Nonetheless, in your video, VO announces the combobox input when you navigate back to it, even though the virtual cursor is visually on the option. You should be able to edit the input text.

The VO cursor needs to be focused on the input in order to enter edit mode,
otherwise pressing other keys of keyboard will be for navigation mode.
However, the easiest way is to close the popover first with escape key and then focus back on the input. 

I am using MacOS Sonoma 14.5.

On Chrome Version 127.0.6533.120 (arm64) also produces the same behavior

https://github.com/user-attachments/assets/0c906ed0-24ee-439e-a8e3-b7b047ddcbc9

I am trying to emulate on MacOS 14.2 with VirtualBuddy and I will report back when finished.

**Comment by Anon-136** [comment]
Emulation on MacOS 14.2

### Results

The VO cursor acts the same on macOS 14.2.
When I try to navigate out of the listbox back to the input, it's going back to the listbox  

https://github.com/user-attachments/assets/a4916d17-b915-4dd0-a220-c6aa22027030

However ScreenReader behavior on macOS 14.2 seems to be different from 14.5
Like you said
> Nonetheless, ... You should be able to edit the input text.

Even though the VO cursor is inside the listbox, I can still edit the input.

But in macOS 14.5 I can't edit the text anymore since it's in navigation mode somehow.

**Comment by diegohaz** [comment]
> However ScreenReader behavior on macOS 14.2 seems to be different from 14.5 Like you said
> 
> > Nonetheless, ... You should be able to edit the input text.
> 
> Even though the VO cursor is inside the listbox, I can still edit the input.
> 
> But in macOS 14.5 I can't edit the text anymore since it's in navigation mode somehow.

Yeah, I'm not sure what I'm doing differently, but I can now reproduce this behavior on macOS 14.2 (where you can edit the input). I'll look into it.

**Comment by diegohaz** [comment]
You can work around this by resetting the combobox active id when the combobox popover receives keyboard focus:

```jsx
function ComboboxPopover(props) {
  const combobox = Ariakit.useComboboxContext();
  return (
    <Ariakit.ComboboxPopover
      {...props}
      // TODO: Remove when https://github.com/ariakit/ariakit/issues/4036 is fixed
      onFocusVisible={(event) => {
        props.onFocusVisible?.(event);
        combobox?.setActiveId(null);
      }}
    />
  );
}
```

**Comment by diegohaz** [comment]
I tried to reproduce this on macOS 26.4.1 in both Chrome 148 and Safari 26.4, but I can’t reproduce it anymore. It may have been fixed in the browsers. I’ll close this issue for now, but if it still happens, please update the reproduction steps. Thank you!
